copy.go

```diff
@@ -54,6 +56,41 @@ type CopyOptions struct {
 	MapRoot func(ctx context.Context, src content.Storage, root ocispec.Descriptor) (ocispec.Descriptor, error)
 }
 
+// selectPlatform implements platform filter and returns the descriptor of
+// the first matched manifest from the manifest list / image index
+func (o *CopyOptions) selectPlatform(ctx context.Context, src content.Storage, root ocispec.Descriptor, p *ocispec.Platform) (ocispec.Descriptor, error) {
+	if root.MediaType == docker.MediaTypeManifestList || root.MediaType == ocispec.MediaTypeImageIndex {
+		manifests, err := content.Successors(ctx, src, root)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
+
+		// platform filter
+		for _, m := range manifests {
+			matched := platform.MatchPlatform(m.Platform, p)
+			if matched {
+				return m, nil
+			}
+		}
+		return ocispec.Descriptor{}, errdef.ErrNotFound
+	}
+	return ocispec.Descriptor{}, errdef.ErrNotFound
```

Comment:
Suggest to change the returning to indent less lines


```suggestion
	if root.MediaType != docker.MediaTypeManifestList && root.MediaType != ocispec.MediaTypeImageIndex {
	    return ocispec.Descriptor{}, errdef.ErrNotFound
	}
	
        manifests, err := content.Successors(ctx, src, root)
        if err != nil {
	        return ocispec.Descriptor{}, err
        }
        
        // platform filter
        for _, m := range manifests {
	        matched := platform.MatchPlatform(m.Platform, p)
	        if matched {
		        return m, nil
	        }
        }
        return ocispec.Descriptor{}, errdef.ErrNotFound
```